src/kafka/producer_consumer.clj

```diff
@@ -0,0 +1,119 @@
+(ns kafka.producer-consumer
+  (:gen-class)
+  (:import
+    (java.time Duration)
+    (java.net InetAddress)
+    (org.apache.kafka.common.errors WakeupException)
+    (org.apache.kafka.common.serialization StringSerializer StringDeserializer)
+    (org.apache.kafka.clients.producer KafkaProducer ProducerRecord)
+    (org.apache.kafka.clients.consumer KafkaConsumer)))
+
+(def producer-properties
+  {"client.id",         (.getHostName (InetAddress/getLocalHost))
+   "group.id",          "group1"
+   "bootstrap.servers", "localhost:9092"
+   "key.serializer"     StringSerializer
+   "value.serializer"   StringSerializer})
+
+(def consumer-properties
+  {"client.id",         (.getHostName (InetAddress/getLocalHost))
+   "group.id",          "group1"
+   "bootstrap.servers", "localhost:9092"
+   "key.deserializer"   StringDeserializer
+   "value.deserializer" StringDeserializer})
+
+(defn build-producer
+  [producer-properties]
+  (KafkaProducer. producer-properties))
+
+(defn build-consumer
+  [consumer-properties]
+  (KafkaConsumer. consumer-properties))
+
+(defn new-msg
+  [topic key msg]
+  (println "Msg created successfully.")
+  (ProducerRecord. topic key msg))
+
+(defn send-msg!
+  [producer msg-value]
+  (let [producer-topic "example-topic"
+        msg            (new-msg producer-topic "msg" msg-value)]
+    (println "Msg sent successfully!")
+    (.send producer msg)
+    (.close producer)))
+
+(defn consumer-subscribe!
+  [consumer topic]
+  (.subscribe consumer [topic])
+  (.commitAsync consumer))
+
+(defn receive-msg-single-thread
+  [consumer topic]
+  (consumer-subscribe! consumer topic)
+  (let [records (.poll consumer (Duration/ofMillis 100000))]
+    (mapv #(str "Msg: " (.value %)) records)))
```

Comment:
1. Let bindings should have the return value as their last element. Keep other code in the `[]`
2. Why is the msg value being polluted with additional strings?
3. Use threading here
4. avoid # function forms (for now)